presto-main/src/test/java/com/facebook/presto/operator/TestPagesIndex.java

```diff
@@ -59,9 +60,33 @@ public void testEstimatedSize()
                 estimatedSizeAfterCompact));
     }
 
-    private static PagesIndex newPagesIndex(List<Type> types, int expectedPositions)
+    @Test
+    public void testEagerCompact()
+    {
+        List<Type> types = ImmutableList.of(VARCHAR);
+
+        PagesIndex lazyCompactPagesIndex = newPagesIndex(types, 50, false);
+        PagesIndex eagerCompactPagesIndex = newPagesIndex(types, 50, true);
+
+        for (int i = 0; i < 5; i++) {
+            lazyCompactPagesIndex.addPage(somePage(types));
+            eagerCompactPagesIndex.addPage(somePage(types));
+        }
+
+        for (int i = 0; i < 5; i++) {
+            Block lazyCompactBlock = lazyCompactPagesIndex.getChannel(0).get(i);
+            Block eagerCompactBlock = eagerCompactPagesIndex.getChannel(0).get(i);
+            // VARCHAR sequence pages can be compacted
+            assertTrue(eagerCompactBlock.getRetainedSizeInBytes() < lazyCompactBlock.getRetainedSizeInBytes(), "Expect retained size of block in eager compact PagesIndex smaller than lazy compact PagesIndex");
+
+            Block compactBlock = lazyCompactBlock.copyRegion(0, lazyCompactBlock.getPositionCount());
```

Comment:
Just `lazyCompactPagesIndex.compact()` and then again compare `PagesIndex.getEstimatedSize()` ?

Reply:
I mean, instead of analyzing `eagerCompactPagesIndex` block by block do something slightly more high-level:

1. `i=0..4`  add a page; check that eager compacting PI has smaller size than lazy compacting
2. force compact in lazy compacting and check that both have same size